specs/validator/0_beacon-chain-validator.md

```diff
@@ -221,19 +221,28 @@ epoch_signature = bls_sign(
 
 ##### Eth1 Data
 
-`block.eth1_data` is a mechanism used by block proposers vote on a recent Ethereum 1.0 block hash and an associated deposit root found in the Ethereum 1.0 deposit contract. When consensus is formed, `state.eth1_data` is updated, and validator deposits up to this root can be processed. The deposit root can be calculated by calling the `get_deposit_root()` function of the deposit contract using the post-state of the block hash.
-
-* Let `D` be the list of `Eth1DataVote` objects `vote` in `state.eth1_data_votes` where:
-    * `vote.eth1_data.block_hash` is the hash of an Eth 1.0 block that is (i) part of the canonical chain, (ii) >= `ETH1_FOLLOW_DISTANCE` blocks behind the head, and (iii) newer than `state.eth1_data.block_hash`.
-    * `vote.eth1_data.deposit_count` is the deposit count of the Eth 1.0 deposit contract at the block defined by `vote.eth1_data.block_hash`.
-    * `vote.eth1_data.deposit_root` is the deposit root of the Eth 1.0 deposit contract at the block defined by `vote.eth1_data.block_hash`.
-* If `D` is empty:
-    * Let `block_hash` be the block hash of the `ETH1_FOLLOW_DISTANCE`'th ancestor of the head of the canonical Eth 1.0 chain.
-    * Let `deposit_root` and `deposit_count` be the deposit root and deposit count of the Eth 1.0 deposit contract in the post-state of the block referenced by `block_hash`
-    * Let `best_vote_data = Eth1Data(block_hash=block_hash, deposit_root=deposit_root, deposit_count=deposit_count)`.
-* If `D` is nonempty:
-    * Let `best_vote_data` be the `eth1_data` member of `D` that has the highest vote count (`D.count(eth1_data)`), breaking ties by favoring block hashes with higher associated block height.
-* Set `block.eth1_data = best_vote_data`.
+The `block.eth1_data` field is for block proposers to vote on recent Eth 1.0 data. This recent data contains an Eth 1.0 block hash as well as the associated deposit root (as calculated by the `get_deposit_root()` method of the deposit contract) and deposit count after execution of the corresponding Eth 1.0 block. If over half of the block proposers in the current Eth 1.0 voting period vote for the same `eth1_data` then `state.eth1_data` updates at the end of the voting period. Each deposit in `block.body.deposits` must verify against `state.eth1_data.eth1_deposit_root`.
```

Comment:
I think it is important to note here that `eth1_data` is only updated in times of finality, particularly because not much is said about `eth1_data` in `0_beacon_chain.md`. Speaking with the guys from RV, they initially missed this fact and it lead to some concerns about safety.

Reply:
It is not only updated in times of finality. 

I was mistaken yesterday that incoming deposits are only activate at times of finality, but even this is not true. I'd leave as is for now, but will think about the scenario discussed yesterday a bit more.

Reply:
Trying to grok [`process_registry_updates`](https://github.com/ethereum/eth2.0-specs/blob/dev/specs/core/0_beacon-chain.md#registry-updates) about the activations at finailty.

regardless. eth1data is updated whenever possible. The thing that might protect against the scenario discussed yesterday will have to do with activations.

Reply:
> it lead to some concerns about safety

What concerns of safety?

Reply:
Can discuss offline. Nothing that should be addressed in this PR